src/PFRICH_geo_v1.cpp

```diff
@@ -0,0 +1,1407 @@
+ //  std::vector<double> rmins = {rmin2, rmin2, rmin1, rmin1, rmin2, rmin2};
+ //  std::vector<double> rmaxs = {rmax, rmax, rmax, rmax, rmax, rmax};
+ //  std::vector<double> zs    = {-length2 / 2., -length1 / 2., -length1 / 2., length1 / 2., length1 / 2., length2 / 2.};
+
+   std::vector<double> rmins = {100,  100};
+   std::vector<double> rmaxs = {1300, 1300};
+   std::vector<double> zs    = {-5000, 5000};
+
+
+ /*--------------------------------------------------*/
+ /*--------------------------------------------------*/
+   Volume     motherVol = description.pickMotherVolume(sdet);
+ //  Material  air      = description.air();
+ //
+ //  Polycone ptube(0.0, 2.0 * M_PI, rmins, rmaxs, zs);
+ //  // Tube ptube(rmin1,rmax,length2);
+ //  Volume envelope(det_name, ptube, air);
+ //
+ //  PlacedVolume env_phv = motherVol.placeVolume(envelope);
+ //  env_phv.addPhysVolID("system", det_id);
+ //  sdet.setPlacement(env_phv);
+
+   sens.setType("tracker");
+   description.invisible();
+
+   xml::DetElement       detElem = e;
+   std::string           detName = detElem.nameStr();
+   xml::Component        dims    = detElem.dimensions();
+
+ //  double vessel_radius   =  dims.attr<double>(_Unicode(rmax));;
+
+   ///*--------------------------------------------------*/
+   /// Porting in GDML volume
+
+   xml_dim_t  x_gdml(x_det.child(_U(gdmlFile)));
+   xml_dim_t  x_par(x_det.child(_U(parent)));
+
+   string     name         = x_det.nameStr();
+   string     gdml         = x_gdml.attr<string>(_U(ref));
+   string     par_nam        = x_par.nameStr();
+   DetElement det_parent   = description.detector(par_nam);
+
+   TGDMLParse parser;
+   if (!gdml.empty() && gdml[0] == '/') {
+     TUri uri(gdml.c_str());
+     gdml = uri.GetRelativePart();
+   } else {
+     string path = xml::DocumentHandler::system_path(e, gdml);
+     TUri   uri(path.c_str());
+     gdml = uri.GetRelativePart();
+   }
+
+   if (!det_parent.isValid()) {
+     except(name, "+++ Cannot access detector parent: %s", par_nam.c_str());
+   }
+
+ //  DetElement sdet(name, id);
+   Volume     volume = parser.GDMLReadFile(gdml.c_str());
+   if (!volume.isValid()) {
+     except("ROOTGDMLParse", "+++ Failed to parse GDML file:%s", gdml.c_str());
+   }
+   volume.import(); // We require the extensions in dd4hep.
+
+   cout << volume.name() << endl;
+
+
+   Volume       mother = det_parent.volume();
+   PlacedVolume pv;
+
+   string     gdml_physvol = dd4hep::getAttrOrDefault<string>(x_gdml, _Unicode(physvol), "");
+
+   if (!gdml_physvol.empty()) {
+     PlacedVolume node = volume->FindNode(gdml_physvol.c_str());
+     if (!node.isValid()) {
+       printout(ERROR, "ROOTGDMLParse", "+++ Invalid gdml placed volume %s", gdml_physvol.c_str());
+       printout(ERROR, "ROOTGDMLParse", "+++ Valid top-level nodes are:");
+       volume->PrintNodes();
+       except("ROOTGDMLParse", "+++ Failed to parse GDML file:%s for node:%s", gdml.c_str(), gdml_physvol.c_str());
+
+       cout << "aaaa" << endl;
```

Comment:
Remove all indiscriminate output. Use `printout` with a debugging level for output.

Reply:
Fixed